File: 1/ClosestValueBST.java
class ClosestValueBST {
	public static int closestVal(BST bst, int value) {
		BST.Node root = bst.getRoot();		
		int closest = root.value;

		while(root!= null) {
			if( Math.abs(closest-value) > Math.abs(root.value-value) ) {
						closest = root.value;
			}

			if(value < root.value) {
					root = root.left;
			} else if(value > root.value) {		
					root = root.right;
			} else {
					return value;
			}
		}

		return closest;
	}

	public static void main(String args[]) {
		BST bst = new BST();
		int value = 12;

		bst.insert(new int[]{10, 5, 15, 2, 6, 13, 22, 14});


		System.out.println( closestVal(bst, value) ); 
	}
	
}

File: 1/js/src/ClosestValueBST.ts
import {BSTNode, BST} from "./BST";

function getClosestValue(bst: BST, data: number): number {
	let currentNode: BSTNode | null = bst.root;
	let closest = Number.POSITIVE_INFINITY;

	while(currentNode != null) {
		if(Math.abs(closest - data) > Math.abs(currentNode.data - data)) {
			closest = currentNode.data;
		}	

		if(data < currentNode.data) {
			currentNode = currentNode.left;
		} else if(data > currentNode.data) {
			currentNode = currentNode.right;
		} else {
			return data;
		}
	}

	return closest;
}

(function main() {
	const bst = new BST();
	const data = 12;

	bst.insertMany([10, 5, 15, 2, 6, 13, 22, 14]);

	console.log(getClosestValue(bst, data));
})();
